WordPress/Classes/ViewRelated/Post/PageAutoUploadMessageProvider.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+
+class PageAutoUploadMessageProvider: AutoUploadMessageProvider {
+
+    // MARK: - Cancelling
+
+    static let changesWillNotBePublished = NSLocalizedString("We won't publish these changes.",
```

Comment:
I liked this approach a lot!

One small suggestion, some of the messages are very similar. Eg.: _"We won't publish these changes."_

I wonder if we should just extend the `AutoUploadMessageProvider` and provide that as a default. This will drastically reduce the size of `PageAutoUploadMessageProvider`  and `PostAutoUploadMessageProvider`.

What do you think?

Reply:
The thing is that the messages are the same only coincidentally.  I think this is one of those cases where redundancy makes sense.

More to the point on why I think this is good redundancy: if the message changed to be different across posts and pages ("We won't publish the changes in this post.") we'd be forced to modify the class structure instead of just changing the string.

Reply:
Additionally, although unlikely, this gives the opportunity for translations to use a specific wording.

Reply:
Hmm, I agree.